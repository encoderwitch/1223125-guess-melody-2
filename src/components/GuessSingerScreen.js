import React from 'react';
import PropTypes from 'prop-types';

const GuessSingerScreen = (props) => {
  const {time, errors, question, onAnswer, screenIndex} = props;

  return (
    <article id="game-artist">
      <section className="game game--artist">
        <header className="game__header">
          <a className="game__back" href="#">
            <span className="visually-hidden">Сыграть ещё раз</span>
            <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию"></img>
          </a>

          <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">
            <circle className="timer__line" cx="390" cy="390" r="370"
              style={{filter: `url(#blur)`, transform: `rotate(-90deg) scaleY(-1)`, transformOrigin: `center`}} />
          </svg>

          <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
            <span className="timer__mins">{time}</span>
            <span className="timer__dots">:</span>
            <span className="timer__secs">00</span>
          </div>
          <div className="game__mistakes">
            <div className="wrong">{errors}</div>
            <div className="wrong">{errors}</div>
            <div className="wrong">{errors}</div>
          </div>
        </header>

        <section className="game__screen">
          <h2 className="game__title">Кто исполняет эту песню?</h2>
          <div className="game__track">
            <div className="track">
              <button className="track__button track__button--play" type="button"></button>
              <div className="track__status">
                <audio></audio>
              </div>
            </div>
          </div>

          <form className="game__artist">
            {question.answers.map((answer) =>
              <div className="artist" key={`${screenIndex}-answer-${answer.id}`}>
                <input
                  id={`answer-${answer.id}`}
                  value={answer.singer}
                  className="artist__input visually-hidden"
                  type="radio"
                  name="answer"
                  onChange={(evt) => {
                    onAnswer(evt.target.value);
                  }}
                ></input>
                <label className="artist__name" htmlFor={`answer-${answer.id}`}>
                  <img className="artist__picture" src={`img/${answer.picture}`} alt={answer.singer}></img>
                  {answer.singer}
                </label>
              </div>
            )}
          </form>
        </section>
      </section>
    </article>
  );
};

GuessSingerScreen.propTypes = {
  time: PropTypes.number,
  errors: PropTypes.number.isRequired,
  question: PropTypes.shape({
    type: PropTypes.oneOf([`singer`]).isRequired,
    song: PropTypes.shape({
      src: PropTypes.string.isRequired,
      singer: PropTypes.string.isRequired
    }).isRequired,
    answers: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      picture: PropTypes.string.isRequired,
      singer: PropTypes.string.isRequired
    }))
  }).isRequired,
  onAnswer: PropTypes.func.isRequired,
  screenIndex: PropTypes.number.isRequired
};

export default GuessSingerScreen;
