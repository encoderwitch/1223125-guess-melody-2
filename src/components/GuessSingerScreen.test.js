
import React from 'react';
import renderer from 'react-test-renderer';
import GuessSingerScreen from './GuessSingerScreen';

const mockQuestion = {
  type: `singer`,
  song: {
    singer: `Biagio Antonacci`,
    src: `biagio_antonacci_-_mio_fratello_feat_mario_incudine_(zf.fm).mp3`
  },
  answers: [
    {
      id: 1,
      picture: `ilvolo.jpg`,
      singer: `Il Volo`
    },
    {
      id: 2,
      picture: `biagio.jpg`,
      singer: `Biagio Antonacci`
    },
    {
      id: 3,
      picture: `adriano.jpg`,
      singer: `Adriano Celentano`
    }
  ]
};

function createNodeMock(element) {
  if (element.type === `audio`) {
    return {createRef() {}};
  }
  return null;
}

it(`GuessSingerScreen correctly renders`, () => {
  const options = {createNodeMock};
  const tree = renderer
    .create(<GuessSingerScreen
      question={mockQuestion}
      time={0}
      errors={0}
      screenIndex={0}
      onAnswer={() => {}}
    />, options)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
