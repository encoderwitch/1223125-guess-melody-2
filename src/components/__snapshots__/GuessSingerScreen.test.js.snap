// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessSingerScreen correctly renders 1`] = `
<article
  id="game-artist"
>
  <section
    className="game game--artist"
  >
    <header
      className="game__header"
    >
      <a
        className="game__back"
        href="#"
      >
        <span
          className="visually-hidden"
        >
          Сыграть ещё раз
        </span>
        <img
          alt="Угадай мелодию"
          className="game__logo"
          src="img/melody-logo-ginger.png"
        />
      </a>
      <svg
        className="timer"
        viewBox="0 0 780 780"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          className="timer__line"
          cx="390"
          cy="390"
          r="370"
          style={
            Object {
              "filter": "url(#blur)",
              "transform": "rotate(-90deg) scaleY(-1)",
              "transformOrigin": "center",
            }
          }
        />
      </svg>
      <div
        className="timer__value"
        xmlns="http://www.w3.org/1999/xhtml"
      >
        <span
          className="timer__mins"
        >
          0
        </span>
        <span
          className="timer__dots"
        >
          :
        </span>
        <span
          className="timer__secs"
        >
          00
        </span>
      </div>
      <div
        className="game__mistakes"
      >
        <div
          className="wrong"
        >
          0
        </div>
        <div
          className="wrong"
        >
          0
        </div>
        <div
          className="wrong"
        >
          0
        </div>
      </div>
    </header>
    <section
      className="game__screen"
    >
      <h2
        className="game__title"
      >
        Кто исполняет эту песню?
      </h2>
      <div
        className="game__track"
      >
        <div
          className="track"
        >
          <button
            className="track__button track__button--play"
            type="button"
          />
          <div
            className="track__status"
          >
            <audio />
          </div>
        </div>
      </div>
      <form
        className="game__artist"
      >
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-1"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="Il Volo"
          />
          <label
            className="artist__name"
            htmlFor="answer-1"
          >
            <img
              alt="Il Volo"
              className="artist__picture"
              src="img/ilvolo.jpg"
            />
            Il Volo
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-2"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="Biagio Antonacci"
          />
          <label
            className="artist__name"
            htmlFor="answer-2"
          >
            <img
              alt="Biagio Antonacci"
              className="artist__picture"
              src="img/biagio.jpg"
            />
            Biagio Antonacci
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-3"
            name="answer"
            onChange={[Function]}
            type="radio"
            value="Adriano Celentano"
          />
          <label
            className="artist__name"
            htmlFor="answer-3"
          >
            <img
              alt="Adriano Celentano"
              className="artist__picture"
              src="img/adriano.jpg"
            />
            Adriano Celentano
          </label>
        </div>
      </form>
    </section>
  </section>
</article>
`;
